### ISSUE TITLE
[ADAP-962] [Bug] Write to BigQuery silently fails

### ISSUE THREAD

**Issue by t-bennett243** [dbt-labs/dbt-bigquery#984]
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I'm running dbt python jobs using DataProc serverless and BigQuery, using the default dataproc batch runtime configuration. When running a simple model (below) with a large number of rows the model fails on write to bigquery with no error messaging.

```python
# models/python_model.py
def model(dbt, spark):
    NUM = 10**7

    data = [(x, f"{x}_as_string") for x in range(NUM)]

    schema = StructType(
        [
            StructField("id", IntegerType(), True),
            StructField("id_string", StringType(), True),
        ]
    )

    df = spark.createDataFrame(data=data, schema=schema)

    return df
```

There are warnings in dataproc logs, however both the dataproc batch and the dbt run are successful even though the table was never created.

```
WARN BigQueryDataSourceWriterInsertableRelation: It seems that 2 out of 0 partitions have failed, aborting
WARN BigQueryDirectDataSourceWriterContext: BigQuery Data Source writer 8319ca09-e791-42a0-be75-7d0899cee63b aborted
```

### Expected Behavior

The dbt job should fail since the table never got created in BigQuery.

### Steps To Reproduce

dbt run --select python_model

### Relevant log output

```shell
Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

### Environment

```markdown
- OS:
- Python: 3.10
- dbt-core: 1.6.5
- dbt-bigquery: 1.6.5
```

**Comment by dbeatty10** [comment]
Thanks for raising this @t-bennett243 !

I wonder if this is an underlying issue with the way that BigQuery is handling this. i.e., should BigQuery be raising an error here instead of just warning?
```
WARN BigQueryDirectDataSourceWriterContext: BigQuery Data Source writer 8319ca09-e791-42a0-be75-7d0899cee63b aborted
```

Do either you or @nickozilla have any suggestions of what dbt-bigquery can do differently here?

**Comment by t-bennett243** [comment]
From a quick search of the spark-bigquery-connector (with my limited Java understanding!), `BigQueryDirectDataSourceWriterContext` looks like just a warning however`BigQueryDataSourceWriterInsertableRelation` [should throw an exception?](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/ef2655ad5dc6c738bc60e7d6100ff06794ce6643/spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/write/BigQueryDataSourceWriterInsertableRelation.java#L120)

We noticed that the write method has changed from "direct" to "indirect" in a recent release of [dbt-bigquery](https://github.com/dbt-labs/dbt-bigquery/pull/681/files) however. This seems to fix this issue for me

**Comment by dbeatty10** [comment]
Good find on https://github.com/dbt-labs/dbt-bigquery/pull/681 !

> This seems to fix this issue for me

Since that PR will be included in the final release of dbt-bigquery 1.7 and it seems to fix the issue for you, I'm going to close this as resolved.

If it turns out to **not** be resolved for you, just reach out and we can either re-open this issue or a new one.
